great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -1395,24 +1394,38 @@ def save_expectation_suite(
         expectation_suite: ExpectationSuite,
         expectation_suite_name: Optional[str] = None,
         overwrite_existing: bool = True,
+        include_rendered_content: Optional[bool] = None,
         ge_cloud_id: Optional[str] = None,
-        **kwargs,
-    ):
+        **kwargs: dict,
+    ) -> None:
         """Save the provided expectation suite into the DataContext.
 
         Args:
-            expectation_suite: the suite to save
-            expectation_suite_name: the name of this expectation suite. If no name is provided the name will \
-                be read from the suite
+            expectation_suite: The suite to save.
+            expectation_suite_name: The name of this Expectation Suite. If no name is provided, the name will be read
+                from the suite.
+            overwrite_existing: Whether to overwrite the suite if it already exists.
+            include_rendered_content: Whether to save the prescriptive rendered content for each expectation.
+            ge_cloud_id: Cloud ID for saving expectation suite.
 
         Returns:
             None
         """
+        if include_rendered_content is None:
+            if (
+                self.include_rendered_content.expectation_suite is True
+                or self.include_rendered_content.globally is True
+            ):
+                include_rendered_content = True
+            else:
+                include_rendered_content = False
+
```

Comment:
Should we make this a private helper? Seems like we use this logic a decent bit

Reply:
It is used across about 6 different modules. We can move it into a util file, but we have identified util files as something to avoid as well.

Reply:
I was stuck on util files, but this should be able to be refactored into `AbstractDataContext`.

Reply:
I don't want to block progress, this may be appropriate here but in general I think we tend to over-use inheritance for code de-duplication.

Inheritance should primarily be used for polymorphism, not code re-use.
https://www.arhohuttunen.com/misconceptions-about-code-reuse/